src/Microsoft.AspNet.Http/Features/FormFeature.cs

```diff
@@ -77,17 +85,32 @@ public IFormCollection ReadForm()
                 throw new InvalidOperationException("Incorrect Content-Type: " + _request.ContentType);
             }
 
+            // TODO: Avoid Sync-over-Async http://blogs.msdn.com/b/pfxteam/archive/2012/04/13/10293638.aspx
```

Comment:
Can you file a bug for this? Or put the bug number in the comment if you already filed one.